### PR TITLE
[#478] Run tests in the Vert.x event loop thread

### PR DESCRIPTION
**Pull request by DavideD**
All the tests extending BaseReactiveTest will now run in the Vert.x event loop Thread by default.

Because the schema generation is blocking I had to delegate the session factory to a Vert.x work pool.

### CODE REVIEW

**Review by Sanne — APPROVED** [review]

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/vertx/VertxInstance.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
  * <pre>{@code
  * new ReactiveServiceRegistryBuilder()
  *     .applySettings( properties )
- *     .addService( VertxInstance.class, () -> myVertx )
+ *     .addService( VertxInstance.class, (VertxInstance) () -> myVertx )
```
Comment (gavinking) [inline hibernate-reactive-core/src/main/java/org/hibernate/reactive/vertx/VertxInstance.java:27]: @DavideD what's going on here? Why is this typecast now necessary?

Comment (DavideD) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/vertx/VertxInstance.java:27]: When I tried this code I had a compilation error and that was the only way of making it work.
Did I miss something?

Comment (gavinking) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/vertx/VertxInstance.java:27]: Oh so you're saying it was always necessary and my Javadoc code example was wrong?

Comment (DavideD) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/vertx/VertxInstance.java:27]: Yep

Comment (gavinking) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/vertx/VertxInstance.java:27]: OK. NP

**Pull request merged**